libs/utils/utils.js

```diff
@@ -873,12 +875,40 @@ const getMepValue = (val) => {
   return finalVal;
 };
 
+const getMdValue = (key) => {
+  const value = getMetadata(key);
+  if (value) {
+    return getMepValue(value);
+  }
+  return false;
+};
+
+const getPromoMepEnablement = () => {
+  const mds = [
+    'apac_manifestnames',
+    'emea_manifestnames',
+    'americas_manifestnames',
+    'jp_manifestnames',
+    'manifestnames',
+  ];
+  const mdObject = mds.reduce((obj, key) => {
+    const val = getMdValue(key);
+    if (val) {
+      obj[key] = val;
+    }
+    return obj;
+  }, {});
+  if (Object.keys(mdObject).length) {
+    return mdObject;
+  }
+  return false;
+};
```

Comment:
Is this logic needed in multiple places? Is it needed pre-LCP? Is there any other place is can exist? We plan on cleaning up utils in the near future to trim it down to the bare minimum.

Reply:
it is indeed needed pre-lcp like all mep related stuff in that file, orientating future rendering of the page based on metadata & query parameters. There is a dedicated mep utils test file. May be we should have a mep-prelcp one, but yeah, that would be an additional pre-lcp call.